### PR TITLE
[GUIDE] Add bulk guide

### PR DESCRIPTION
**Pull request by zethuman**
### Description
_Add bulk guide._

### Issues Resolved
_Closes [#278, #279]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
You're missing DCO, same as other PRs.

**Comment by zethuman** [comment]
fixed

File: guides/bulk.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+import (
+    "github.com/opensearch-project/opensearch-go/v2"
+    "log"
+)
+
+func main() {
+    client, err := opensearch.NewDefaultClient()
+    if err != nil {
+        log.Printf("error occurred: [%s]", err.Error())
```
Comment (dblock) [inline guides/bulk.md:20]: Should we bailing here in samples, e.g. `log.Fatalf`?

**Review by dblock — APPROVED** [review]

File: guides/bulk.md (hunk 1 above)
Comment (zethuman) [reply guides/bulk.md:20]: I was guided by the rules from [issue](https://github.com/opensearch-project/opensearch-go/issues/278#issue-1654505429), there was written not to throw errors, or did I misunderstand?

Comment (dblock) [reply guides/bulk.md:20]: Maybe, I don't know what's best. 

What's the user experience when `NewDefaultClient()` returns an error with the code as written?

Comment (zethuman) [reply guides/bulk.md:20]: In the console there will be a reason why the client was not created and the program will complete successfully, I mean there will be no fatal termination. The user, having read the reason, understands that he did something wrong, and decides the origin reason and tries again.

Comment (dblock) [reply guides/bulk.md:20]: In this example the code doesn't do anything else, however in the example below:

```go
movies := "movies"
books := "books"
createMovieIndex, err := client.Indices.Create(movies)
if err != nil {
    log.Printf("error occurred: [%s]", err.Error())
}
log.Printf("response: [%+v]", createMovieIndex)
createBooksIndex, err := client.Indices.Create(books)
if err != nil {
    log.Printf("error occurred: [%s]", err.Error())
}
log.Printf("response: [%+v]", createBooksIndex)
```

You get an error the first time on  `client.Indices.Create(movies)`, then you get an error again the second time on `client.Indices.Create(books). I believe we shouldn't be bailing after the first error, no?

Comment (zethuman) [reply guides/bulk.md:20]: I agree about the first one, but again I will refer to the guide ... It's not difficult for me to change it for a fatal mistake. At the expense of the second, at the end of each guide there are cleaning topics, and the previously created modules were cleaned there

**Pull request merged**